src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -206,13 +209,17 @@ internal ModelParameters(IReadOnlyList<IReadOnlyList<WordItemScore>> wordScoresP
             }
         }
 
-        [BestFriend]
-        internal ModelParameters GetLdaDetails(int iinfo)
+        /// <summary>
+        /// Method to provide details about the topic discovered by LightLDA
+        /// </summary>
+        /// <param name="columnIndex">index of column</param>
+        /// <returns></returns>
+        public ModelParameters GetLdaDetails(int columnIndex)
         {
-            Contracts.Assert(0 <= iinfo && iinfo < _ldas.Length);
+            Contracts.Assert(0 <= columnIndex && columnIndex < _ldas.Length);
```

Comment:
**suggestion**: So I don't know how usable this API would be for users, specially because users might not be sure of what index corresponds to the column they want. Is it possible to have `public ModelParameters GetLdaDetails(string columnName)` instead? (throw an exception if the provided name can't be matched with the names found in _columns)

Reply:
The order is same order as user define column options pair in below API. In most cases this should be 0 as user is using lda by input column, output column pair.
Maybe I can rename this as "index of column options pair" to make it clear

        /// <include file='doc.xml' path='doc/members/member[@name="LightLDA"]/*' />
        /// <param name="env">The environment.</param>
        /// <param name="columns">Describes the parameters of the LDA process for each column pair.</param>
        internal LatentDirichletAllocationEstimator(IHostEnvironment env, params ColumnOptions[] columns)
        {
            Contracts.CheckValue(env, nameof(env));
            _host = env.Register(nameof(LatentDirichletAllocationEstimator));
            _columns = columns.ToImmutableArray();
        }


---
In reply to: [448138812](https://github.com/dotnet/machinelearning/pull/5260#discussion_r448138812) [](ancestors = 448138812)

Reply:
I see. I think rephrasing the docs of `GetLdaDetails` to be "Index of the column options pair" would be better.

---
In reply to: [448520738](https://github.com/dotnet/machinelearning/pull/5260#discussion_r448520738) [](ancestors = 448520738,448138812)